src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IEnumerableDefaultConverter.cs

```diff
@@ -22,6 +22,7 @@ protected override bool OnWriteResume(Utf8JsonWriter writer, TCollection value,
             if (state.Current.CollectionEnumerator == null)
             {
                 enumerator = value.GetEnumerator();
+                state.Current.CollectionEnumerator = enumerator;
```

Comment:
Are these three converters the only ones accessing the `CollectionEnumerator` field?

Reply:
No, I have now aligned all of them to the new pattern even if their enumerator is not inheriting from `IDisposable`.